frappe/defaults.py

```diff
@@ -85,18 +85,19 @@ def get_user_permissions(user=None):
 
 
 def get_defaults(user=None):
-	globald = get_defaults_for()
+	global_defaults = get_defaults_for()
 
 	if not user:
 		user = frappe.session.user if frappe.session else "Guest"
 
-	if user:
-		userd = {}
-		userd.update(get_defaults_for(user))
-		userd.update({"user": user, "owner": user})
-		globald.update(userd)
+	if not user:
+		return global_defaults
```

Comment:
This line never reaches, since L91 always sets a truthy value to `user`

Reply:
What if `frappe.session` is truthy but `frappe.session.user` is `None`?

Reply:
Not sure when this can happen, anyway lets leave it for now.